common/src/test/java/io/netty/util/concurrent/DefaultPromiseTest.java

```diff
@@ -449,6 +453,177 @@ public void mustPassNullContextToContextListener() {
         promise.setSuccess(result);
     }
 
+    @Test
+    public void mapMustApplyMapperFunctionWhenFutureSucceeds() {
```

Comment:
nit: Maybe we should extract these tests to a different class since they are not really associated with `DefaultPromise` any more.